test/jdk/java/net/httpclient/HttpResponseConnectionLabelTest.java

```diff
@@ -0,0 +1,442 @@
+
+    // Secondary server-client pairs
+
+    private static final ServerRequestPair SEC_HTTP1 = ServerRequestPair.of(Version.HTTP_1_1, false);
+
+    private static final ServerRequestPair SEC_HTTPS1 = ServerRequestPair.of(Version.HTTP_1_1, true);
+
+    private static final ServerRequestPair SEC_HTTP2 = ServerRequestPair.of(Version.HTTP_2, false);
+
+    private static final ServerRequestPair SEC_HTTPS2 = ServerRequestPair.of(Version.HTTP_2, true);
+
+    private static SSLContext createSslContext() {
+        try {
+            return new SimpleSSLContext().get();
+        } catch (IOException exception) {
+            throw new UncheckedIOException(exception);
+        }
+    }
+
+    private record ServerRequestPair(
+            HttpTestServer server,
+            HttpRequest request,
+            boolean secure,
+            CountDownLatch[] serverResponseLatchRef) {
+
+        private static final AtomicInteger SERVER_COUNTER = new AtomicInteger();
+
+        private static final AtomicInteger SERVER_RESPONSE_COUNTER = new AtomicInteger();
+
+        private ServerRequestPair {
+            try {
+                server.start();
+            } catch (Exception serverException) {
+                throw new RuntimeException("failed closing server", serverException);
+            }
+        }
+
+        private static ServerRequestPair of(Version version, boolean secure) {
+
+            // Create the server and the request URI
+            SSLContext sslContext = secure ? SSL_CONTEXT : null;
+            HttpTestServer server = createServer(version, sslContext);
+            String handlerPath = "/" + /* salting the path: */ HttpResponseConnectionLabelTest.class.getSimpleName();
+            String requestUriScheme = secure ? "https" : "http";
+            URI requestUri = URI.create(requestUriScheme + "://" + server.serverAuthority() + handlerPath);
+
+            // Register the request handler
+            String serverId = "" + SERVER_COUNTER.getAndIncrement();
+            CountDownLatch[] serverResponseLatchRef = new CountDownLatch[1];
+            server.addHandler(
+                    (exchange) -> {
+                        String responseBody = "" + SERVER_RESPONSE_COUNTER.getAndIncrement();
+                        String connectionKey = exchange.getConnectionKey();
+                        try (exchange) {
+
+                            // Participate in the latch count down
+                            if (serverResponseLatchRef[0] != null) {
+                                serverResponseLatchRef[0].countDown();
+                                LOGGER.log(
+                                        "Server[%s] is waiting for the latch... (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                assertTrue(serverResponseLatchRef[0].await(2, TimeUnit.SECONDS));
```

Comment:
We should avoid using any kind of timeouts here and in other places where we call this `await(...)` in this test. Given how varied the test execution environments are and our past experience with such timeouts, there's no right timeout to choose from. Instead, we should just do a `await()` without the timeout, so that if for whatever reason the latch isn't counted down, then the jtreg test execution timeout (which is controlled and configured externally to the test) will kick in and jtreg will do the necessary work of failing the test and also gathering detailed diagnostics through any failure handlers that are configured in that test execution environment.

Reply:
Removed timeouts in 5b418fd7a9f4cfc3da940611ae3ba9d43e57aa91.